### ISSUE TITLE
How to adjust hyperparameters when PiSSA initialization underperforms while vanilla LoRA + Kaiming works well?

### ISSUE THREAD

**Issue by UnbearableFate** [GraphPKU/PiSSA#48]
Hello, and thank you for releasing PiSSA.
I am currently evaluating PiSSA initialization on Llama-3.1-8B for a commonsense-reasoning fine-tuning task. Under the same hyperparameter configuration, I consistently observe the following behavior:
Vanilla LoRA with Kaiming initialization trains stably and reaches good validation accuracy.
PiSSA initialization with HuggingFace, however, produces significantly weaker performance in the early steps and does not recover even over multiple epochs.

This suggests that the hyperparameters that work well for Kaiming-initialized LoRA may not directly transfer to PiSSA. I would appreciate guidance on how the training configuration should be adapted for PiSSA, or what adjustments are typically required to make PiSSA behave stably.

⸻

Current training configuration

"${PYTHON_PATH}" finetune.py \
    --base_model=meta-llama/Llama-3.1-8B \
    --data_path=/LLM-Adapters/ft-training_set/commonsense_170k.json \
    --output_dir=./outputs \
    --batch_size=64 \ # global bs
    --per_device_train_batch_size=4 \
    --num_epochs=3 \
    --learning_rate=2e-4 \
    --warm_step=300 \
    --lr_scheduler_type=cosine \
    --weight_decay=0.01 \
    --cutoff_len=512 \
    --val_set_size=1024 \
    --eval_step=100 \
    --save_step=100 \
    --adapter_name=lora \
    --target_modules="[\"q_proj\",\"k_proj\",\"v_proj\",\"o_proj\",\"gate_proj\",\"up_proj\",\"down_proj\"]" \
    --lora_r=32 \
    --lora_alpha=64 \
    --bf16 \
    --init_lora_weights=pissa \
    --enable_torch_compile

Questions
	1.	Are there recommended learning-rate or α–r scaling adjustments specific to PiSSA?
For example, should PiSSA generally use a lower LR (e.g., 5e-5–1e-4) or a different α/r ratio?
	2.	Should weight decay, warmup ratio, or optimizer choice differ from vanilla LoRA when using PiSSA?
	3.	Is PiSSA more sensitive to batch size or bf16 numerics during the early updates?

Any troubleshooting guidance, recommended hyperparameter ranges, or example training configurations would be extremely helpful.

Thank you in advance for your time and clarification.

<img width="974" height="623" alt="Image" src="https://github.com/user-attachments/assets/84b1ac47-d352-4109-80a5-9e5665375d28" />

<img width="892" height="648" alt="Image" src="https://github.com/user-attachments/assets/59bf1cee-d70f-465c-9a38-b79062c5507f" />

Train and eval loss figure, where green line is pissa and magenta one is kaiming initialization.

**Comment by UnbearableFate** [comment]
I solved this problem.

**Comment by Jack-Rickwood** [comment]
How did you solve it?
